gui/src/components/Nfts.tsx

```diff
@@ -0,0 +1,134 @@
+import {
+  Box,
+  CircularProgress,
+  ImageList,
+  ImageListItem,
+  ImageListItemBar,
+} from "@mui/material";
+import { invoke } from "@tauri-apps/api/tauri";
+import React, { useEffect, useState } from "react";
+import InfiniteScroll from "react-infinite-scroller";
+
+import { useNetworks, useWallets } from "../store";
+import { Address, NftToken, Paginated, Pagination } from "../types";
+import { Modal, NftDetailsView } from "./";
+import { AddressView } from "./AddressView";
+import { Panel } from "./Panel";
+
+export function Nfts() {
+  const account = useWallets((s) => s.address);
+  const chainId = useNetworks((s) => s.current?.chain_id);
+
+  const [detailsViewOpen, setDetailsViewOpen] = useState(false);
+  const [currentNftDetails, setCurrentNftDetails] = useState<
+    NftToken | undefined
+  >(undefined);
+  const [pages, setPages] = useState<Paginated<NftToken>[]>([]);
+
+  const loadMore = () => {
+    let pagination: Pagination = {};
+    const last = pages?.at(-1)?.pagination;
+    if (!!last) {
+      pagination = last;
+      pagination.page = (pagination.page || 0) + 1;
+    }
+
+    invoke<NftToken>("db_get_nft_tokens", {
```

Comment:
instead of a direct db query here, we should create a `store/nfts.ts` to manage all the state